app/src/main/java/org/wikipedia/page/PageToolbarHideHandler.java

```diff
@@ -56,12 +56,13 @@
      */
     void setFadeEnabled(boolean enabled) {
         fadeEnabled = enabled;
-        update();
+
+        // to avoid the toolbar applies the opacity before image loaded
+        pageFragment.getWebView().postDelayed(this::update, DELAY_UPDATE_MS);
```

Comment:
When I see code that uses `postDelayed` to work around a problem, it usually means that there's an incomplete understanding of the underlying issue. (see below)